design/space_views.md

```diff
@@ -0,0 +1,152 @@
+  * Need to figure out how this is constrained compared to Spatial 2D
+
+
+#### On the idempotence of 2D & 3D Space Views
+In the early prototype 2D and 3D Space Views were separate since they would use different
+renderers - 3D Space Views were driven by `three-d`, 2D Space Views by egui directly.
+With the advent or `re_renderer`, this distinction was no longer necessary and indeed a hindrance.
+Like most modern renderer, `re_renderer` does not distinguish 2D and 3D rendering at a fundamental level
+(albeit we might add some limited awareness in the future) since shader, hardware acceleration and
+data structures are all fundamentally the same.
+
+If the root of a 2D Space View has a camera projection we can have a defined way of displaying any 3D content.
+Therefore, all 3D content can be displayed in a 2D Space View.
+
+Vice versa, if an entity in a 3D space defines camera intrinsics, any 2D contents under it can be previewed
+in 3D space. Again, there is no point in putting a limit on what is displayed there.
+
+Arguments for distinguishing 2D and 3D views anyways:
+* Control of the `Eye` is vastly different
+* Most users have expectations on what a 2D view looks and feels
+  * Heuristics of how things are displayed by default may vary wildly
+    * e.g. a depth image in 3D should be displayed as a point cloud whenever possible, whereas in 2D this makes rarely sense
+* 2D and 3D may have different properties regarding grid lines, background and rendering effects (?)
+  * the later is doubtful since again 3D content may be displayed in 2D views
+
+
+Note that the strong overlaps between 2D and 3D views constrain re-usability points in the design
+of `SpaceViewClass`
+
+
+### Space View State
+In addition to blueprint stored data, a space view has a class specific `SpaceViewState`
+which stored ephemeral state that is not persisted as part of the blueprint.
+This is typically used for animation/transition state.
+
+⚠️ As of writing, we're using this also for state that *should* be be persisted and needs to be moved to
+blueprint components.
+
+
+### Scene
+Scenes are created every frame from the data blueprint in order to draw the view.
+Each Space View Class defines which scene parts and scene context parts it needs to create its scene.
+
+#### `ScenePart`
+A scene part defines how a **single given archetype** is processed during the scene buildup.
+As part of its population it may set arbitrary temporary internal state and emit re_renderer drawables.
+
+#### `SceneContextPart`
+Similar to `ScenePart` but does not emit drawables. Accessible while scene parts are populated.
+This is used e.g. to prepare the transform tree.
+
+#### Scene Lifecycle
+Each frame, each `SpaceViewClass` builds up a scene. The framework defines a fixed lifecycle for all views.
+Given a `SpaceViewClass` `MyClass`:
+* default instantiate a new `TypedScene<MyClass>`
+  * this contains an instance of ``MyClass::SceneParts` and `MyClass::SceneParts::Context`,
```

Comment:
I thinnk this would resolve itself with rename and further restructure of `SpaceViewFrame`. Frankly though I have a hard time doing this restructure in text form only since I can't be sure if everything sticks then. I can iterate on the code and come back to this then

Reply:
If editing the code first is what works for you, definitely do that -- but once you've got it in a good state I think having a clean overview of the intended relationships is going to be essential for making sense of the diff and understanding whether the modified code adheres to the intention of the design or not.